sled-agent/src/hardware/non_illumos/mod.rs

```diff
@@ -0,0 +1,37 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use slog::Logger;
+use std::sync::Mutex;
+use tokio::sync::broadcast;
+
+struct HardwareInner {
+    is_scrimlet: bool,
+}
+
+/// A simulated representation of the underlying hardware.
+///
+/// This is intended for non-illumos systems to have roughly the same interface
+/// as illumos systems.
+pub struct Hardware {
+    log: Logger,
+    inner: Mutex<HardwareInner>,
```

Comment:
Is this Mutex actually needed? It looks like scrimlet config is only set during initialization.